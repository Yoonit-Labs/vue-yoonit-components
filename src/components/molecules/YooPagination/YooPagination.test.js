import { shallowMount } from '@vue/test-utils'
import yooPagination from '@/components/molecules/YooPagination/YooPagination.vue'
import PropsConfig from '@/components/molecules/YooPagination/YooPagination.config'

const SlotText = 'Default Slot Text'

const mountComponent = () => {
  return shallowMount(yooPagination, {
    slots: { default: SlotText }
  })
}

describe('YooPagination Component', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mountComponent()
  })

  it('Matches Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Loads the Component HTML', () => {
    expect(wrapper.classes('yoo__container--yooPagination')).toBe(true)
    expect(wrapper.find('yooPagination.yoo-btn').exists()).toBe(true)
  })

  /* describe('Props', () => {
    describe('animation', () => {
      it('Has a valid default value', () => {
        expect(PropsConfig.animation.options.includes(yooPagination.props.animation.default)).toBe(true)
      })
      PropsConfig.animation.options.forEach(animation => {
        it('Include animation class: ', async () => {
          await wrapper.setProps({ animation })
          expect(wrapper.find('').classes().includes(`--animation-${animation}`)).toBe(true)
        })
      })
    })
  }) */ // describe Props

  describe('Events', () => {
    describe('Click', () => {
      it('Emits Click Event', async () => {
        expect(wrapper.emitted()).toHaveProperty('last')
        expect(wrapper.emitted()).toHaveProperty('tapPage')
      })
    })
  })

  describe('Events', () => {
    describe('Click', () => {
      it('Emits Click Event', async () => {
        expect(wrapper.emitted()).toHaveProperty('tapPage')
      })
    })
  }) // describe Events
})
