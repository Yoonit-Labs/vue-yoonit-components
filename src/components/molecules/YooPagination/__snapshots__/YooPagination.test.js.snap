// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooPagination Component Matches Snapshot 1`] = `
<yooflexlayout-stub flexdirection="row" flexwrap="nowrap" justifycontent="space-between" alignitems="stretch" aligncontent="stretch" gap="0">
  <yoobutton-stub variation="outline" buttonsize="normal" fill="primary" icon="arrow-left" iconposition="left" iconsize="normal" iconstyle="solid" animation="none" text="" iconcustomcolor="" class="button__container--back"></yoobutton-stub>
  <yoostepper-stub countertype="dot" totalsteps="1" currentstep="0"></yoostepper-stub>
  <yoobutton-stub variation="base" buttonsize="normal" fill="primary" icon="arrow-right" iconposition="left" iconsize="normal" iconstyle="solid" animation="none" text="" iconcustomcolor=""></yoobutton-stub>
  <yoobutton-stub variation="base" buttonsize="normal" fill="primary" icon="arrow-right" iconposition="right" iconsize="normal" iconstyle="solid" animation="none" text="" iconcustomcolor="" flexgrow="1" style="display: none;"></yoobutton-stub>
  <yoobutton-stub variation="base" buttonsize="normal" fill="primary" icon="arrow-right" iconposition="right" iconsize="normal" iconstyle="solid" animation="none" text="" iconcustomcolor="" flexgrow="1" style="display: none;"></yoobutton-stub>
</yooflexlayout-stub>
`;
